pkg/kv/kvprober/kvprober.go

```diff
@@ -306,11 +308,42 @@ func (p *Prober) Metrics() Metrics {
 	return p.metrics
 }
 
+// LogLeaseholderInfo searches through the given tracing recording for log messages
+// indicating updated range information, extracts range IDs and leaseholder node IDs,
+// and logs this information only if the leaseholder for a range has been updated.
+func (p *Prober) LogLeaseholderInfo(ctx context.Context, recording tracingpb.Recording) {
+	// Find the log message containing information about updated ranges
+	informationLog, isValid := recording.FindLogMessage("received updated range")
+
+	if isValid {
+		// Regular expressions to extract range IDs and leaseholder node IDs
+		rangeRegex := regexp.MustCompile(`desc: r(\d+)`)
+		leaseRegex := regexp.MustCompile(`repl=\(n(\d+)`)
+
+		// Extract all occurrences of range IDs and leaseholder node IDs from the log message
+		ranges := rangeRegex.FindAllStringSubmatch(informationLog, -1)
+		leaseholders := leaseRegex.FindAllStringSubmatch(informationLog, -1)
```

Comment:
Not 100% confident, but I think in the case of the prober, we expect exactly one leaseholder to be involved, since only a single KV pair in KV is probed. When you run your prototype, do you see multiple leaseholders in the logs or one?

Reply:
Yes, the line I have used to get information seems to sometimes get information for the ranges just before and after the range being probed as shown in the comment below.